Practical Example/IosInterviewPracticeProject/IosInterviewPracticeProjectTests/CourseAPIServiceTests/CourseAPIServiceTests.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  NetworkManagerTests.swift
+//  IosInterviewPracticeProject
+//
+//  Created by Robert Ramirez on 8/20/20.
+//  Copyright © 2020 me.robert.ramirez. All rights reserved.
+//
+
+import XCTest
+@testable import raywenderlich
+
+class CourseAPIServiceTests: XCTestCase {
+  var artworkUrl: URL?
+  var courseAPIService = CourseAPIService()
+
+  override func setUpWithError() throws {
+    artworkUrl = URL(string: "https://files.betamax.raywenderlich.com/attachments/collections/244/e2d20345-06fb-42f4-8366-c7ffe80b5f77.png")
+  }
+
+    override func tearDownWithError() throws {
+      artworkUrl = nil
+    }
+
+  func testFetchArticles() {
+    let expect = expectation(description: "Fetch Articles")
+    let articleUrl = "https://raw.githubusercontent.com/raywenderlich/ios-interview/master/Practical%20Example/articles.json"
+    var items = [Item]()
+
+    courseAPIService.fetchContent(for: articleUrl) { results in
+      switch results {
+      case .success(let results):
+        items = results.data
+        expect.fulfill()
+      case .failure(let error):
+        XCTFail(error.localizedDescription)
+      }
+    }
+    waitForExpectations(timeout: 3, handler: nil)
+    XCTAssertGreaterThan(items.count, 0, "Failed to fetch articles")
+  }
+
+  func testFetchVideos() {
+    let expect = expectation(description: "Fetch Videos")
+    let videoUrl = "https://raw.githubusercontent.com/raywenderlich/ios-interview/master/Practical%20Example/videos.json"
```

Comment:
It's probably not a good idea to connect to the internet and fetch actual data in your tests. There are lots of logistical reasons why (it's slow, you might DDOS your server, etc.), but there's one important more philosophical one. Unit tests verify that _your code_ works as intended. Your test should fail if there's a mistake in your code. If the server is down, the unit test probably shouldn't fail. Here you're not just testing your code but the server itself. Think about using mocks and stubs to simulate successful and failed loads.

Reply:
If you have any resources I could look at, I would appreciate it.